code/game/objects/objs.dm

```diff
@@ -93,9 +93,9 @@
 
 
 /obj/attack_ghost(mob/user)
+	if(ui_interact(user) != -1)

```

Comment:
This returns -1 if What was clicked on wasn't a Nanoui. Basically this prevents both the attack_ai proc and the interact proc being called at the same time.